mypy/checker.py

```diff
@@ -2923,6 +2926,16 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
 
         Handle all kinds of assignment statements (simple, indexed, multiple).
         """
+
+        if isinstance(s.rvalue, TempNode) and s.rvalue.no_rhs:
+            for lvalue in s.lvalues:
+                if (
+                    isinstance(lvalue, NameExpr)
+                    and isinstance(var := lvalue.node, Var)
+                    and isinstance(instance := get_proper_type(var.type), Instance)
```

Comment:
It seems to me this will miss cases where a deprecated class appears in a nested position, e.g. `foo: Deprecated | Other` or `bar: list[Deprecated]`. No need to fix in this PR, you can open a follow up PR, or issue.